File: backend/src/main/java/com/simats/neoomfs/dto/request/MedicalHistoryRequest.java
package com.simats.neoomfs.dto.request;

import lombok.Data;

/**
 * Inbound DTO for medical history (Step 5 of the wizard).
 */
@Data
public class MedicalHistoryRequest {
    private boolean hypertension;
    private boolean diabetes;
    private boolean heartDisease;
    private boolean kidneyDisease;
    private boolean liverDisease;
    private boolean thyroidDisorder;
    private boolean asthma;
    private boolean epilepsy;
    private boolean bloodDisorder;
    private boolean hepatitis;
    private boolean hivPositive;
    private boolean pregnant;
    private String pregnancyTrimester;
    private String otherConditions;
    private String currentMedications;
    private String allergies;
    private String previousSurgeries;
    private String anaestheticComplications;
    private String familyHistory;
    private String socialHistory;
    private String notes;
}

File: frontend/app/src/main/java/com/simats/neoomfs/models/MedicalHistoryRequest.kt
package com.simats.neoomfs.models

data class MedicalHistoryRequest(
    val hypertension: Boolean,
    val diabetes: Boolean,
    val heartDisease: Boolean,
    val kidneyDisease: Boolean,
    val liverDisease: Boolean,
    val thyroidDisorder: Boolean = false,
    val asthma: Boolean,
    val epilepsy: Boolean = false,
    val bloodDisorder: Boolean,
    val hepatitis: Boolean = false,
    val hivPositive: Boolean = false,
    val pregnant: Boolean,
    val pregnancyTrimester: String? = null,
    val otherConditions: String? = null,
    val currentMedications: String? = null,
    val allergies: String? = null,
    val previousSurgeries: String? = null,
    val anaestheticComplications: String? = null,
    val familyHistory: String? = null,
    val socialHistory: String? = null,
    val notes: String? = null
)
